test/TestProjects/TypeSchemaMapping/Customized/ModelWithArrayOfEnum.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using NamespaceForEnums;
+
+namespace TypeSchemaMapping.Models
+{
+    /// <summary> The ModelWithArrayOfEnum. </summary>
+    internal partial class ModelWithArrayOfEnum
+    {
+        public IReadOnlyList<EnumForModelWithArrayOfEnum> ArrayOfEnum { get; }
```

Comment:
I would also declare a list (or RO list) of a nullable extensible enum or something because, in my bug, I noticed that the de/serialization code didn't always consistently check for null; e.g., during write, it just called `x.Value.ToString()`, IIRC. If not set, that would fail. In the serialization code, an extensible enum was created with `null` which would actually throw. So while, based on the sample at least, it seems my blocking problem is fixed, I don't think the whole bug is resolved.

Reply:
...or maybe you or I should open another bug on *that* behavior (bad de/serialization of actual nullable types) to move past this one. In my case, none of my value types are nullable (or, more specifically, array items).

Reply:
I'll open a new bug.

Reply:
https://github.com/Azure/autorest.csharp/issues/798